source/common/network/address_impl.cc

```diff
@@ -175,8 +176,8 @@ std::string Ipv4Instance::sockaddrToString(const sockaddr_in& addr) {
 absl::uint128 Ipv6Instance::Ipv6Helper::address() const {
   absl::uint128 result{0};
   static_assert(sizeof(absl::uint128) == 16, "The size of asbl::uint128 is not 16.");
-  memcpy(static_cast<void*>(&result), static_cast<const void*>(&address_.sin6_addr.s6_addr),
-         sizeof(absl::uint128));
+  memcpy(static_cast<void*>(&result), // NOLINT(safe-memcpy)
```

Comment:
sorry, one more idea. Should we just make a wrapper unsafeMemcpy for the case where neither source are safe, rather than using NOLINT everywhere's? That's just an aesthetic suggestion. Your choice.

But in this case I don't see why we couldn't use:
```
safeMemcpyUnsafeSrc(&result, &address_.sin6_addr.s6_addr);
```
Also this was pre-existing but I don't see why result needs to be initialized to 0 above. We are going to be fully writing all the bits with the memcpy.

Reply:
Regarding the use of a macro from memcpy, I completely agree that ```NOLINT``` just adds more clutter to the code  😄. However, the check_format.py script allows exemptions based on the use of ```NOLINT```, hence it seems more logical to keep it that way, right?

About your second question, I think we had already talked about it, but I am not able to make the copy without the ```static_cast```, since result is a class, not just an array, so the compiler complains as shown below.

```
error: 'void* memcpy(void*, const void*, size_t)' copying an object of type 'class absl::uint128' with 'private' member 'absl::uint128::lo_' from an array of 'const uint8_t [16]' {aka 'const unsigned char [16]'}; use assignment or copy-initialization instead [-Werror=class-memaccess]
   20 | #define safeMemcpyUnsafeSrc(dst, src) memcpy(dst, src, sizeof(*(dst)));
      |                                                                      ^
```

Reply:
RIght -- the idea here is that you'd still use `NOLINT` in the implementation of `safeMemcpyUnsafeSrc`, but that seeing `safeMemcpyUnsafeSrc` at call-sites would be a little more format-friendly, and also serve as a more explicit reminder to the reviewer that they need to check the bounds.

I think leaving the static cast for the array is OK, but could you also try using `&address_.sin6_addr.s6_addr[0]` without the cast? I don't think you need to cast `&result` -- anything should convert to a `void*` without an explicit cast.

Reply:
I could replace the above with:
```
absl::uint128 Ipv6Instance::Ipv6Helper::address() const {
  static_assert(sizeof(absl::uint128) == 16, "The size of asbl::uint128 is not 16.");
  uint64_t result_lo, result_hi;
  safeMemcpyUnsafeSrc(&result_lo, &address_.sin6_addr.s6_addr[0]);
  safeMemcpyUnsafeSrc(&result_hi, &address_.sin6_addr.s6_addr[8]);
  return absl::MakeUint128(result_hi, result_lo);
}
```

However, I do not know if it is the right byte order. WDYT?

Reply:
Why can't you use
```
uint128_t result;
safeMemcpyUnsafeSrc(&result, &address_.sin6_addr.s6_addr[0]);
```